### PR TITLE
Feature fixed sidenav

### PR DESCRIPTION
**Pull request by lyssareba**
# Description
This pull request is meant to allow the SideNav to be fixed as well as show the SideNav trigger on Large Displays. This should fix Issue #416  

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update (Add showOnLarge to SideNav PropTypes
https://react-materialize.github.io/#/sidenav)

# How Has This Been Tested?

Using the example SideNav in the docs, I used create-react-app to build a test application. I imported and initiated the SideNav. After passing a fixed prop to SideNav, the navigation stayed fixed and the trigger is not visible. If a trigger is passed with fixed, it will display the trigger on mobile only. While passing the showOnLarge prop, I can now see the trigger on Large Displays as well as mobile. 

All tests were passing without adjustment. I added an extra test for the showOnLarge prop. 

I also removed the isRequired on trigger prop. The only problem here is, there is no error if you do not pass a trigger. To resolve this, I added a console.warn that lets the developer know there was no trigger passed to SideNav.

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have not generated a new package version. (the maintainers will handle that)

### CODE REVIEW

**Comment by alextrastero** [comment]
Also instead of a warning I would add more info to the proptypes, that way when creating the docs this will be parsed and added to the proptype table

**Comment by lyssareba** [comment]
@alextrastero 
I added the showOnLarge prop just to make it easier for styling. It applies the hide-on-large-only if showOnLarge is not applied to maintain the materialize behavior. It also allows an easier flow from NavBar if you want to use NavBar on desktop and a SideNav on mobile.

I can remove those additions if you would prefer users just add the class names on the trigger. If that's the preferred method, it may be useful to include the common class names for those functionalities in an example. 

As for the proptypes, I'm not sure how to add the information you are referring too. Is there documentation or an example you could link? I'm only 6ish months into React.

**Comment by alextrastero** [comment]
I see your point, the only issue is that we can't add a new prop for each scenario. Ideally we would have a generic solution that would work with other materialize-css helper classes. 
I do like however the idea that if there's no trigger the side nav should be fixed.

As for the docs, you need to add a comment in the propTypes section as other components have :)

https://github.com/react-materialize/react-materialize/blob/master/src/Carousel.js#L71

**Comment by alextrastero** [comment]
By the way, I just merged a `prettier` PR so you might need to fixed some conflicts. Let me know if you need some help with that.

**Comment by lyssareba** [comment]
My bad on the propTypes, I did add showOnLarge in my first commit in the propTypes.
```
  /**
   * Adds showOnLarge class to Trigger
   */
  showOnLarge: PropTypes.bool,
```

I didn't realize you were referring to those. 

I could adjust the SideNav to manage that logic itself. 

- If no trigger is passed, it automatically applies the fixed class.
- If a trigger is passed with fixed, it will apply the fixed class on desktop, but will hide the sideNav and display the trigger on med-and-down.
- If a trigger is passed without fixed, it will show on all screen sizes.

Is it too much to assume those use cases are the main issues?

**Comment by alextrastero** [comment]
Let's give it a try! I would separate the trigger logic into a `renderTrigger` method so that it's easier to understand what's going on :)

Good luck!

**Comment by lyssareba** [comment]
@alextrastero Everything should be done here.
Fixed prettier issues, and resolved that logic to not use showOnLarge. 
Shouldn't be any merge conflicts or test issues at this point.

File: src/SideNav.js [hunk 1]
```diff
@@ -15,11 +15,12 @@ class SideNav extends Component {
   }
 
   render() {
-    const { className, children, fixed, ...props } = this.props;
+    const { className, children, ...props } = this.props;
     delete props.id;
     delete props.trigger;
     delete props.options;
-
+    delete props.fixed;
```
Comment (alextrastero) [inline src/SideNav.js]: the way deconstruct works is that if you add `fixed` to L18 then you don't need to delete it from the `props` object

Comment (alextrastero) [reply src/SideNav.js]: also add trigger to L18 and delete it from L20

File: src/SideNav.js [hunk 2]
```diff
@@ -15,11 +15,12 @@ class SideNav extends Component {
...
+    const fixed = this.props.fixed || !this.props.trigger;
     const classNames = cx('side-nav', { fixed }, className);
```
Comment (alextrastero) [inline src/SideNav.js]:
```jsx
const classNames = cx('side-nav', className, {
  fixed: fixed || !trigger
});
```

File: test/SideNavSpec.js [hunk 3]
```diff
@@ -46,6 +46,11 @@ describe('<SideNav />', () => {
     assert.isTrue(sideNav.hasClass('side-nav fixed red'));
   });
 
+  it('should render trigger on large screens if `showOnLarge`', () => {
```
Comment (alextrastero) [inline test/SideNavSpec.js]: let's also add some specs to check the `fixed` logic and what happens when there's no `trigger`

File: src/SideNav.js (hunk 2 above)
Comment (lyssareba) [reply src/SideNav.js]: Didn't realize I could shorten this, this way. Thanks :)

**Comment by Danny-G-Smith** [comment]
I would love to use this feature, however I am not sure where to place the props.  Best I can tell the documentation at https://react-materialize.github.io/#/sidenav didn't get updated.  Any help would be appreciated.

**Comment by alextrastero** [comment]
Hey @Danny-G-Smith, to render a fixed sidenav just pass `fixed` as a prop ;)

**Pull request merged**